src/bootstrap/tool.rs

```diff
@@ -162,7 +162,8 @@ impl Step for ToolBuild {
                 "the following dependencies are duplicated although they \
                       have the same features enabled:"
             );
-            for (id, cur, prev) in duplicates.drain_filter(|(_, cur, prev)| cur.2 == prev.2) {
+            duplicates.retain(|(_, cur, prev)| cur.2 != prev.2);
+            for (id, cur, prev) in &duplicates {
```

Comment:
Please double check that I got this conversion right -- drain_filter and retain appear to require inverse return values. But maybe I'm misreading something.